src/ol/interaction/selectinteraction.js

```diff
@@ -149,12 +149,23 @@ ol.interaction.Select = function(opt_options) {
   this.filter_ = options.filter ? options.filter :
       ol.functions.TRUE;
 
+  var style = options.style;
```

Comment:
i think it would quite inconsistent with the other interactions to accept an ol.FeatureStyleFunction instead of an ol.style.StyleFunction, even if that would that would make the following lines unnecessary.